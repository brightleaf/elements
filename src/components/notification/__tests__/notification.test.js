import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
} from '@testing-library/react'
import { toBeInTheDocument, toHaveClass } from '@testing-library/jest-dom'
import { Notification } from '../index'

expect.extend({ toBeInTheDocument, toHaveClass })
describe('Column and Columns', () => {
  afterEach(cleanup)
  it('should render', () => {
    const { container } = render(
      <Notification>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render isInfo with class is-info', () => {
    const { container } = render(
      <Notification isInfo>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-info')
  })
  /*
     isMobile,
  className,
  isFluid,
  isWideScreen,
  isFullHD,
  */
  it('should render isSuccess with class is-success', () => {
    const { container } = render(
      <Notification isSuccess>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-success')
  })
  it('should render isDanger with class is-danger', () => {
    const { container } = render(
      <Notification isDanger>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-danger')
  })
  it('should render isPrimary with class is-primary', () => {
    const { container } = render(
      <Notification isPrimary>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-primary')
  })
  it('should render isLink with class is-link', () => {
    const { container } = render(
      <Notification isLink>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-link')
  })
  it('should render isWarning with class is-warning', () => {
    const { container } = render(
      <Notification isWarning>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('is-warning')
  })
  it('should not render with when isShow is false', async () => {
    const mockDismiss = jest.fn(() => {})
    const { container, getByTitle } = render(
      <Notification isShown isPrimary onDismiss={mockDismiss}>
        <div>Hi</div>
      </Notification>
    )
    expect(container.firstChild).toMatchSnapshot()

    fireEvent.click(getByTitle(/dismiss notification/i))
    const notificationNode = await waitForElement(() =>
      getByTitle('notification element')
    )
    expect(notificationNode).toMatchSnapshot()
    expect(mockDismiss).toHaveBeenCalled()
  })
})
