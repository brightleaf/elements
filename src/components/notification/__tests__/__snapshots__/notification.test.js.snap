// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column and Columns should not render with when isShow is false 1`] = `
<div
  class="notification animated animation fadeIn is-primary"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should not render with when isShow is false 2`] = `
<div
  class="notification animated animation fadeOutLeft is-primary"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render 1`] = `
<div
  class="notification animated animation fadeIn"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render isDanger with class is-danger 1`] = `
<div
  class="notification animated animation fadeIn is-danger"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render isInfo with class is-info 1`] = `
<div
  class="notification animated animation fadeIn is-info"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render isLink with class is-link 1`] = `
<div
  class="notification animated animation fadeIn is-link"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render isPrimary with class is-primary 1`] = `
<div
  class="notification animated animation fadeIn is-primary"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render isSuccess with class is-success 1`] = `
<div
  class="notification animated animation fadeIn is-success"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;

exports[`Column and Columns should render isWarning with class is-warning 1`] = `
<div
  class="notification animated animation fadeIn is-warning"
  title="notification element"
>
  <button
    class="delete notification-dismiss"
    title="dismiss notification"
  />
  <div>
    Hi
  </div>
</div>
`;
